### PR TITLE
External account binding

### PR DESCRIPTION
**Pull request by unixcharles**
Based on #219 and #210

Add tests, tweak the interface and readme.

### CODE REVIEW

**Comment by unixcharles** [comment]
@saarons @Shaeli do you mind giving it 👀

File: lib/acme/client.rb [hunk 1]
```diff
@@ -123,7 +136,7 @@ def kid
     @kid ||= account.kid
   end
 
-  def new_order(identifiers:, not_before: nil, not_after: nil)
+  def new_order(identifiers:, not_before: nil, not_after: nil, eab: nil)
```
Comment (saarons) [inline lib/acme/client.rb:139]: Why add `eab` here? I'm not sure I follow.

File: lib/acme/client.rb [hunk 2]
```diff
@@ -50,7 +50,8 @@ def initialize(jwk: nil, kid: nil, private_key: nil, directory: DEFAULT_DIRECTOR
 
   attr_reader :jwk, :nonces
 
-  def new_account(contact:, terms_of_service_agreed: nil)
+  def new_account(contact:, terms_of_service_agreed: nil, eab: nil, external_account_binding: nil)
```
Comment (saarons) [inline lib/acme/client.rb:53]: We could just force people to use `external_account_binding` instead of `eab`, I doubt it would change much.

**Review by saarons — COMMENTED** [review]
Added 2 very minor comments but otherwise looks great to me.

File: lib/acme/client.rb [hunk 3]
```diff
@@ -59,7 +60,19 @@ def new_account(contact:, terms_of_service_agreed: nil)
       payload[:termsOfServiceAgreed] = terms_of_service_agreed
     end
 
-    response = post(endpoint_for(:new_account), payload: payload, mode: :jws)
+    external_account_binding = eab || external_account_binding
+    if external_account_binding
+      kid, hmac_key = external_account_binding.values_at(:kid, :hmac_key)
+      if kid.nil? || hmac_key.nil?
+        raise ArgumentError, 'must specify kid or hmac_key key for external_account_binding'
```
Comment (Shaeli) [inline lib/acme/client.rb:67]:
```suggestion
        raise ArgumentError, 'must specify kid and hmac_key key for external_account_binding'
```

Comment (Shaeli) [inline lib/acme/client.rb:67]:
```suggestion
        raise ArgumentError, 'must specify kid and hmac_key key for external_account_binding'
```

**Review by Shaeli — COMMENTED** [review]
Thanks a lot @unixcharles for adding the tests 🙇‍♀️ 

LGTM 👍 , added a very minor suggestion - feel free to ignore it

File: lib/acme/client.rb (hunk 1 above)
Comment (unixcharles) [reply lib/acme/client.rb:139]: No idea actually. I included it based on #210 but ended up deleting it after looking at [7.3.4 in spec](https://www.rfc-editor.org/rfc/rfc8555.html#section-7.3.4) I don't see any reference to it. I forgot to clean it up the argument.

File: lib/acme/client.rb (hunk 2 above)
Comment (unixcharles) [reply lib/acme/client.rb:53]: I agree

**Pull request merged**